### ISSUE TITLE
[font-metrics-api] FontMetrics IDL is invalid

### ISSUE THREAD

**Issue by Ms2ger** [w3c/css-houdini-drafts#822]
https://drafts.css-houdini.org/font-metrics-api/#fontmetrics-definition has sequence-valued attributes; such attributes are not allowed in IDL.

**Comment by kojiishi** [comment]
@Ms2ger I'm not very familiar with IDL, a novice question. Is what you meant that `advances`, `baselines`, and `fonts` attributes that return `sequence<...>` are not allowed, and they need to be functions, or not to return `sequence<...>`?

**Comment by Ms2ger** [comment]
Indeed. An attribute with a `sequence<…>` return type would by definition return a new `Array` every time you access it, and that's considered a bad API. You can use `FrozenArray`, or make them functions.

**Comment by kojiishi** [comment]
Understood, thank you. I was missing the difference between `sequence` and `FrozenArray`.

We're trying to sync this with Canvas API, and its Canvas API has `FrozenArray`. We should fix this one too to match to the Canvas API.
